### DISCUSSION TITLE
Activities (multiple sets of multiple tags)

### DISCUSSION THREAD

**Discussion by maltalef101** [leftwm/leftwm#1036] [Ideas]
I was wondering if sets of tags could be implemented in a KDE Plasma *Activities* fashion. My guess is that this feature is self explanatory. I find myself using multiple tags for just one purpose and having that kind of separation between *Activities* (be it work and other things) would be very useful.

I am willing to give it a shot at implementing this feature. One thing I have not thought through fully is the keybinds that would be used for such feature, `Win + Alt + [ 0-9]` could be an idea, but I'm gonna need feedback on this.

Thank you very much for reading.
Regards.

**Comment by VuiMuich** [comment]
Tbh it is not as self explanatory to me, as I have never used this feature and my last active running KDE was with slackware 13.x

Would each activity workspace have its own bar instance and set of keybinds? Other then that I currently can't really imagine any benefit. On the other hand I would imagine the mental overhead to keep track of what window is organised where would increase significantly.

If you would like to give it a shot, you are welcome to open a draft PR. If it turns out this wont match well with the project it can go in the contrib repo as a patch, if someone wants to add it to their personal implementation of leftwm.

**Reply by maltalef101** [reply]
> Tbh it is not as self explanatory to me

My bad, sorry.

Activities in Plasma are switchable groups of open windows that, given the floating nature of KWin, have a specified position in space. A better way of naming this feature, and more coherent if you ask me (they like to change the names of things), would be _Virtual Desktops_. I also haven't used KDE for a long time now, so I'm not completely certain of the state of this functionality today.

The way I would adapt this feature, is to have something among the same principle, but instead of switchable groups of windows, you would have groups of tags; each Virtual Desktop has its own set of tags from 0-9, and you can switch between them. You obviously would be able to disable this feature via a `config.ron` option, and you also would be able to define the number of Virtual Desktops you'd want.

I find having a play-work separation useful, and this is a feature I've missed from my days of more _mainstream_ DEs.


> Would each activity workspace have its own bar instance and set of keybinds? 

I mean, you _could_, so why not? I originally did not imagine that but, yeah! 


> If you would like to give it a shot, you are welcome to open a draft PR.

Thanks for the enthusiasm! I've yet to be more akin to Rust and this project in particular, so I'll give it a  chance, and we'll see how it goes. =)

**Reply by VuiMuich** [reply]
This sounds like launching a sub-session with Xephyr would almost fulfill all your desires.
We would just need a way themes to be able to distinguish the native from the virtual session, but I guess that is just a matter of making use of the `DISPLAY` env variable in the up and down script. That again would probably make your theme not the most probable in the sense of sharing it in the community themes for easy drop in support, but it is a very unique tiling wm workflow anyways 🙃

**Reply by maltalef101** [reply]
I wasn't thinking about themes when I originally came up with this idea. Personally, I wouldn't find it useful, but it's a nice feature that someone could need if they wanted to make their virtual desktops more visually distinguishable. I'll try my hand at writing an implementation that is usable at least, and go on from there.
